### PR TITLE
feature - cancel order

### PR DESCRIPTION
**Pull request by at-sonvu**

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Orders/OrdersViewController.swift [hunk 1]
```diff
@@ -79,11 +74,14 @@ class OrdersViewController: ViewController {
             guard let this = self else { return }
             switch result {
             case .success:
-                this.delegate?.viewController(this, needsPerformAction: .didGetDetailOrderSuccess)
                 this.updateView()
-                if this.viewModel.orderState != "new" {
+                if this.viewModel.orderState == "paid" {
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Orders/OrdersViewController.swift]: ennum

File: MyApp/View/Controllers/Orders/OrdersViewController.swift [hunk 2]
```diff
@@ -79,11 +74,14 @@ class OrdersViewController: ViewController {
...
                     this.paidButton.isEnabled = false
                     this.paidButton.alpha = 0.5
+                } else if this.viewModel.orderState == "cancelled" {
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Orders/OrdersViewController.swift]: ennum

File: MyApp/View/Controllers/Orders/OrdersViewController.swift (hunk 1 above)
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Orders/OrdersViewController.swift]: switch case

File: MyApp/View/Controllers/Orders/OrdersViewController.swift [hunk 3]
```diff
@@ -92,6 +90,39 @@ class OrdersViewController: ViewController {
...
+        cancelOrderButton.addSubview(cancelLabel)
+        navigationItem.rightBarButtonItem = UIBarButtonItem(customView: cancelOrderButton)
+    }
+
+    func cancelOrderButtonTouchUpInside() {
+        alert(title: App.Strings.confirm, msg: App.Strings.alertCancelOrder, buttons: [App.Strings.ok, App.Strings.no]) { [weak self] (actions) in
+            guard let title = actions.title, let this = self else { return }
+            switch title {
+            case App.Strings.ok:
+                this.viewModel.processOrder(status: "cancelled", { [weak self](result) in
```
Comment (at-tuanmai2) [inline MyApp/View/Controllers/Orders/OrdersViewController.swift]: enum

**Pull request merged**